app/src/pages/review/[resourceType]/[id].tsx

```diff
@@ -1,28 +1,134 @@
+}
 
 /**
  * Component which renders a page that displays the JSON data of a resource. Also will eventually
  *  provide the user with the ability to make review comments and visualize previous review comments.
  */
 export default function CommentPage() {
+  const ctx = trpc.useContext();
+  const [dateSelected, setDateSelected] = useState(false);
+  const ref = useRef<HTMLInputElement>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
   const router = useRouter();
   const { resourceType: resourceType, id: resourceID, authoring } = router.query;
-  let resource;
+  const { data: resource } = getResource();
+
+  // Validates that all of the necessary form properties were entered by the user
+  const form = useForm({
+    initialValues: {
+      type: '',
+      comment: '',
+      name: '',
+      date: ''
+    },
+    // An error will be thrown if these fields aren't entered properly
+    validate: {
+      type: isNotEmpty('Select the type of comment'),
+      comment: isNotEmpty('Enter artifact comment')
+    }
+  });
+
+  // Currently we can only update draft artifact resources.
+  const resourceUpdate = trpc.draft.updateDraft.useMutation({
+    onSuccess: () => {
+      // This if statement prevents the success notifaction from popping up in the unique case
+      // that the draft artifact has just had an extension array added to the JSON.
+      notifications.show({
+        title: 'Comment Successfully added!',
+        message: `Comment Successfully added to ${resourceType}/${resourceID}`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      ctx.draft.getDraftById.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: 'Update Failed!',
+        message: `Attempt to update ${resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  function getResource() {
+    if (authoring === 'true') {
+      return trpc.draft.getDraftById.useQuery({
+        id: resourceID as string,
+        resourceType: resourceType as ArtifactResourceType
+      });
+    } else {
+      return trpc.service.getArtifactById.useQuery({
+        id: resourceID as string,
+        resourceType: resourceType as ArtifactResourceType
+      });
+    }
+  }
+
+  function parseUpdate(comment: string, type: string, userName: string, dateSelected: boolean) {
+    const additions: DraftArtifactUpdates = {};
+    const deletions: DraftArtifactUpdates = {};
+
+    const newExtensionObject: fhir4.Extension[] = [];
+    newExtensionObject.push({ url: 'type', valueCode: type }, { url: 'text', valueMarkdown: comment });
+
+    if (userName !== '') {
+      newExtensionObject.push({ url: 'user', valueString: userName });
+    }
+    if (dateSelected === true) {
+      const now = new Date();
+      const isoString = now.toISOString();
+      newExtensionObject.push({ url: 'authoredOn', valueDateTime: isoString });
```

Comment:
```suggestion
      const authoredDate = new Date();
      newExtensionObject.push({ url: 'authoredOn', valueDateTime: authoredDate.toISOString() });
```
I think this is a little cleaner/more descriptive!